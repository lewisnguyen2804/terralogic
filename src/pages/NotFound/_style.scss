.notfound-container {
    width: 100vw;
    height: 100vh;
    position: relative;
}

.notfound-centering {
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    .notfound-title {
        color: $main-color;
        @extend .bold-font;
        text-align: center;
    }
    .notfound-return-button {
        // text-align: center;
        padding: 10px 70px;
    }
}

