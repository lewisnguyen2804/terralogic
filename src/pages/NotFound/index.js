import React, { Component } from 'react'
import CustomButton from '../../components/CustomButton';
import { history } from '../../helpers/history';

export default class NotFound extends Component {
    goHome = () => {
        history.push("/")
    }

    render() {
        return (
            <div className="notfound-container">
                <div className="notfound-centering">
                    <div className="notfound-title">
                        <h2>PAGE NOT FOUND</h2>
                    </div>
                    <div className="notfound-return-button">
                        <CustomButton
                            type="button"
                            className="button-type-1 notfound-return-button"
                            onClick={this.goHome}
                            value="HOME" />
                    </div>
                </div>
            </div>
        )
    }
}
