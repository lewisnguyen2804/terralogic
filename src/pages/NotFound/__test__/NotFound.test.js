
import React from 'react'
import { shallow } from 'enzyme';
import NotFound from '../';

describe('test for Login Page', () => {
    const wrapper = shallow(<NotFound />);
    const instance = wrapper.instance();
    const goBackFn = jest.spyOn(instance, 'goHome');
    wrapper.update();
    instance.forceUpdate();
    it('should render correctly', () => {
        const buttonTest = wrapper.find('CustomButton');
        buttonTest.at(0).simulate('click');
        // expect(JSON.stringify(wrapper)).toMatchSnapshot();
        expect(goBackFn).toHaveBeenCalled()
    });
});