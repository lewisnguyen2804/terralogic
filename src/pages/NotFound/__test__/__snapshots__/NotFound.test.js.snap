// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Login Page should render correctly 1`] = `"{}"`;
